src/MainOpt.cxx

```diff
@@ -120,119 +113,121 @@ int MainOpt::parse_json_file(string config_file) {
     return -6;
   }
   vali.Reset();
+  find_broker(*document);
+  find_broker_config(*document);
+  find_conversion_threads(*document);
+  find_conversion_worker_queue_size(*document);
+  find_main_poll_interval(*document);
+  find_brokers_config(*document);
+  find_status_uri(*document);
 
-  {
-    auto &v = d.FindMember("broker")->value;
-    if (v.IsString()) {
-      set_broker(v.GetString());
-    }
-  }
-  {
-    auto &v = d.FindMember("broker-config")->value;
-    if (v.IsString()) {
-      broker_config = {v.GetString()};
-    }
-  }
-  {
-    auto &v = d.FindMember("conversion-threads")->value;
-    if (v.IsInt()) {
-      conversion_threads = v.GetInt();
-    }
-  }
-  {
-    auto &v = d.FindMember("conversion-worker-queue-size")->value;
-    if (v.IsInt()) {
-      conversion_worker_queue_size = v.GetInt();
-    }
-  }
-  {
-    auto &v = d.FindMember("main-poll-interval")->value;
-    if (v.IsInt()) {
-      main_poll_interval = v.GetInt();
-    }
+  return 0;
+}
+
+rapidjson::Document *MainOpt::parse_document(FILE *f1) {
```

Comment:
I suggest moving the `fopen` into this function and passing in the filename string instead of the file stream pointer. Then I think this can be greatly simplified, it is possible to do something like this:
```
#include <rapidjson/document.h>
#include <rapidjson/istreamwrapper.h>
#include <fstream>

using namespace rapidjson; 
using namespace std;

ifstream ifs("test.json");
IStreamWrapper isw(ifs);
Document d;
d.ParseStream(isw);
```